### PR TITLE
[JSON RPC] implement `get_metadata`

### PR DESCRIPTION
**Pull request by christinacelee**
## Motivation

Implementing `get_metadata` JSON RPC method
Also adding `get_latest_commit_metadata` to LibraDB interface, since there is no way to retrieve block timestamp in current interface

## Test Plan

`test_get_metadata`

### CODE REVIEW

File: json-rpc/src/tests.rs [hunk 1]
```diff
@@ -69,44 +72,53 @@ fn test_json_rpc_protocol() {
...
+        all_txns.extend(
+            txns_to_commit
+                .iter()
+                .map(|txn_to_commit| txn_to_commit.transaction().clone()),
+        );
+    }
+
+    (db, all_accounts)
+}
+
```
Comment (thefallentree) [inline json-rpc/src/tests.rs]: Do we just want to an mock db?  This will require write to disk.  which may be too heavy for an unittest.

Comment (christinacelee) [reply json-rpc/src/tests.rs]: yeah - a MockDB doesn't exist right now, so I can make a separate diff for this

File: json-rpc/src/views.rs [hunk 2]
```diff
@@ -26,6 +26,18 @@ impl AccountView {
...
+    version: u64,
+    timestamp: u64,
+}
+
+impl BlockMetadata {
+    pub fn new(version: u64, timestamp: u64) -> Self {
+        BlockMetadata { version, timestamp }
+    }
+}
+
```
Comment (davidiw) [inline json-rpc/src/views.rs]: just make version and timestamp pub

**Pull request closed without merging**